Model/FinalizePayment.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+
+namespace Satispay\Satispay\Model;
+
+
+class FinalizePayment
+{
+    /**
+     * @var \Magento\Sales\Model\Order\Email\Sender\OrderSender
+     */
+    protected $orderSender;
+
+    /**
+     * Order constructor.
+     * @param \Magento\Framework\App\Action\Context $context
+     * @param \Magento\Sales\Model\Order\Email\Sender\OrderSender $orderSender
+     * @param \Satispay\Satispay\Model\Method\Satispay $satispay
+     */
+    public function __construct(
+        \Magento\Framework\App\Action\Context $context,
+        \Magento\Sales\Model\Order\Email\Sender\OrderSender $orderSender,
+        \Satispay\Satispay\Model\Method\Satispay $satispay
+    ) {
+        $this->orderSender = $orderSender;
+    }
+
+    /**
+     * Finalize a Magento 2 Order Payment following the Satispay Payment Data
+     *
+     * @param $satispayPayment
+     * @param \Magento\Sales\Model\Order $order
+     * @return bool
+     */
+    public function finalizePayment($satispayPayment, \Magento\Sales\Model\Order $order)
+    {
+        if ($satispayPayment->status == 'ACCEPTED') {
+            $payment = $order->getPayment();
+            $payment->setTransactionId($satispayPayment->id);
+            $payment->setCurrencyCode($satispayPayment->currency);
+            $payment->setIsTransactionClosed(true);
+            $payment->registerCaptureNotification($satispayPayment->amount_unit / 100, true);
+
+            $order->setState($order::STATE_PROCESSING);
+            $order->setStatus($order::STATE_PROCESSING);
+            $order->save();
+
+            // Payment is OK: send the new order email
+            if (!$order->getEmailSent()) {
+                $this->orderSender->send($order);
+            }
+            return true;
+        } elseif ($satispayPayment->status == 'CANCELED') {
```

Comment:
since you return in the preceding if you don't need the else and it becomes just an if